src/libraries/Hooks.sol

```diff
@@ -58,50 +73,68 @@ library Hooks {
         if (
             permissions.beforeInitialize != self.hasPermission(BEFORE_INITIALIZE_FLAG)
                 || permissions.afterInitialize != self.hasPermission(AFTER_INITIALIZE_FLAG)
-                || permissions.beforeAddLiquidity != self.hasPermission(BEFORE_ADD_LIQUIDITY_FLAG)
-                || permissions.afterAddLiquidity != self.hasPermission(AFTER_ADD_LIQUIDITY_FLAG)
-                || permissions.beforeRemoveLiquidity != self.hasPermission(BEFORE_REMOVE_LIQUIDITY_FLAG)
-                || permissions.afterRemoveLiquidity != self.hasPermission(AFTER_REMOVE_LIQUIDITY_FLAG)
+                || permissions.beforeAddLiquidity != self.hasPermission(BEFORE_ADD_LIQ_FLAG)
+                || permissions.afterAddLiquidity != self.hasPermission(AFTER_ADD_LIQ_FLAG)
+                || permissions.beforeRemoveLiquidity != self.hasPermission(BEFORE_REMOVE_LIQ_FLAG)
+                || permissions.afterRemoveLiquidity != self.hasPermission(AFTER_REMOVE_LIQ_FLAG)
                 || permissions.beforeSwap != self.hasPermission(BEFORE_SWAP_FLAG)
                 || permissions.afterSwap != self.hasPermission(AFTER_SWAP_FLAG)
                 || permissions.beforeDonate != self.hasPermission(BEFORE_DONATE_FLAG)
                 || permissions.afterDonate != self.hasPermission(AFTER_DONATE_FLAG)
+                || permissions.beforeSwapReturnDelta != self.hasPermission(BEFORE_SWAP_RETURNS_DELTA_FLAG)
+                || permissions.afterSwapReturnDelta != self.hasPermission(AFTER_SWAP_RETURNS_DELTA_FLAG)
+                || permissions.afterAddLiqReturnDelta != self.hasPermission(AFTER_ADD_LIQ_RETURNS_DELTA_FLAG)
+                || permissions.afterRemoveLiqReturnDelta != self.hasPermission(AFTER_REMOVE_LIQ_RETURNS_DELTA_FLAG)
         ) {
             revert HookAddressNotValid(address(self));
         }
     }
 
     /// @notice Ensures that the hook address includes at least one hook flag or dynamic fees, or is the 0 address
     /// @param hook The hook to verify
+    /// @return bool True if the hook address is valid
     function isValidHookAddress(IHooks hook, uint24 fee) internal pure returns (bool) {
+        // The hook can only have a flag to return a hook delta if it has the corresponding hook flag
+        if (
+            (!hook.hasPermission(BEFORE_SWAP_FLAG) && hook.hasPermission(BEFORE_SWAP_RETURNS_DELTA_FLAG))
```

Comment:
same here, this could be simplified to 
```
if (!hook.hasPermission(BEFORE_SWAP_FLAG + BEFORE_SWAP_RETURNS_DELTA_FLAG)
```

slightly more readable, tho you'll need to change `hasPermission` to ensure it == flag rather than != 0

Reply:
ooh yeah thats nice

Reply:
wait but this is different though this is to say
!(`BEFORE_SWAP_FLAG`) & `BEFORE_SWAP_RETURNS_DELTA_FLAG`
but your suggestion is
!(`BEFORE_SWAP_FLAG` & `BEFORE_SWAP_RETURNS_DELTA_FLAG`)